### PR TITLE
*: remove all unnecessary mut

### PR DESCRIPTION
**Pull request by choleraehyq**
I use a higher version rustc and it warns me about these `mut`s.
@siddontang @BusyJay PTAL.

Signed-off-by: Cholerae Hu <huyingqian@pingcap.com>

### CODE REVIEW

File: src/coprocessor/codec/mysql/json/json_modify.rs [hunk 1]
```diff
@@ -93,7 +93,7 @@ impl Json {
             if let Json::Object(ref mut map) = *self {
                 if map.contains_key(key) {
                     // e.g. json_replace('{"a": 1}', '$.a', 2) => '{"a": 2}'
-                    let mut v = map.get_mut(key).unwrap();
+                    let v = map.get_mut(key).unwrap();
```
Comment (hicqu) [inline src/coprocessor/codec/mysql/json/json_modify.rs]: `get` instead of `get_mut` is enough?

**Comment by BusyJay** [comment]
LGTM

**Comment by hicqu** [comment]
LGTM

**Comment by choleraehyq** [comment]
Fixed more warnings. @BusyJay PTAL

**Comment by BusyJay** [comment]
Let's merge #2186 first.

**Pull request merged**